pkg/controllers/pricing/suite_test.go

```diff
@@ -297,4 +297,35 @@ var _ = Describe("Pricing", func() {
 		Expect(ok).To(BeTrue())
 		Expect(price).To(BeNumerically("==", 1.10))
 	})
+	It("should update on-demand pricing with response from the pricing API when in the CN partition", func() {
+		tmpPricingProvider := pricing.NewProvider(ctx, awsEnv.PricingAPI, awsEnv.EC2API, "cn-anywhere-1")
+		tmpController := controllerspricing.NewController(tmpPricingProvider)
```

Comment:
Why are we creating a new controller? why can't we use the existing controller used in every other test?

Reply:
A pricing provider will be updated by `pricing.Controller.Reconcile()`. However, the existing `controller` is initialized with the default pricing provider (for `us-west-2`) and cannot be modified:

https://github.com/aws/karpenter-provider-aws/blob/c2016b2dcaf23ff2de6b98e169f320140520948a/pkg/controllers/pricing/suite_test.go#L63

Therefore, I have to create a new `pricing.Controller` to update the `tmpPricingProvider`.